clientimpl.go

```diff
@@ -42,248 +50,359 @@ func NewMazzarothClient(servers []string, options ...Options) (*ClientImpl, erro
 	}
 
-	body, err := json.Marshal(request)
+	return response, nil
+}
+
+// BlockHeaderList calls the endpoint: /v1/channels/{channel_id}/blockheaders?{blockHeight,number}.
+func (c *ClientImpl) BlockHeaderList(channelID string, blockHeight int, number int) (*xdr.Response, error) {
+	url := fmt.Sprintf("%s/%s/channels/%s/blockheaders?height=%d&number=%d", c.serverSelector.Pick(), version, channelID, blockHeight, number)
+
+	response, err := makeRequest(c.httpClient, http.MethodGet, url, nil)
 	if err != nil {
-		return nil, errors.Wrap(err, "unable to create a new request")
+		return nil, errors.Wrap(err, "unable to make a request to blockheaders by blockheight lookup endpoint")
+	}
+
+	return response, nil
+}
+
+// ChannelLookup calls the endpoint: /v1/channels/{channel_id}.
+func (c *ClientImpl) ChannelLookup(channelID string) (*xdr.Response, error) {
+	url := fmt.Sprintf("%s/%s/channels/%s", c.serverSelector.Pick(), version, channelID)
+
+	response, err := makeRequest(c.httpClient, http.MethodGet, url, nil)
+	if err != nil {
+		return nil, errors.Wrap(err, "unable to make a request to channel lookup endpoint")
 	}
 
-	url := c.serverSelector.Pick() + "/abi"
-	req, err := http.NewRequestWithContext(context.Background(), http.MethodPost, url, bytes.NewBuffer(body))
+	return response, nil
+}
+
+// ChannelAbi calls the endpoint: /v1/channels/{channel_id}/abi.
+func (c *ClientImpl) ChannelAbi(channelID string) (*xdr.Response, error) {
+	url := fmt.Sprintf("%s/%s/channels/%s/abi", c.serverSelector.Pick(), version, channelID)
+
+	response, err := makeRequest(c.httpClient, http.MethodGet, url, nil)
 	if err != nil {
-		return nil, errors.Wrap(err, "unable to create a new request")
+		return nil, errors.Wrap(err, "unable to make a request to channel lookup endpoint")
 	}
 
-	req.Header.Set("Content-Type", "application/json")
+	return response, nil
+}
+
+// BlockHeight calls the endpoint: /v1/channels/{channel_id}/blocks/height.
+func (c *ClientImpl) BlockHeight(channelID string) (*xdr.Response, error) {
+	url := fmt.Sprintf("%s/%s/channels/%s/blocks/height", c.serverSelector.Pick(), version, channelID)
 
-	resp, err := c.httpClient.Do(req)
+	response, err := makeRequest(c.httpClient, http.MethodGet, url, nil)
 	if err != nil {
-		return nil, errors.Wrap(err, "unable to make http request")
+		return nil, errors.Wrap(err, "unable to make a request to channel lookup endpoint")
 	}
-	defer resp.Body.Close()
 
-	if resp.StatusCode != http.StatusOK {
-		return nil, errors.New("status code is not OK")
+	return response, nil
+}
+
+// AccountLookup calls the endpoint: /v1/channels/{channel_id}/accounts/{accout_id}.
+func (c *ClientImpl) AccountLookup(channelID string, seed string) (*xdr.Response, error) {
+	seedBin, err := hex.DecodeString(seed)
+	if err != nil {
+		return nil, errors.Wrap(err, "unable to decode seed")
+	}
+	privateKey := ed25519.NewKeyFromSeed(seedBin)
+	address, err := xdr.IDFromPublicKey(privateKey.Public())
```

Comment:
This should not be done.  AccountLookup should just use the provided string as the hex Account ID in the lookup, no need to get a private key.

Reply:
With the single parameter interface version, this problem doesn't exist anymore as the priv key never gets into the function. Thanks